pandas/_libs/tslib.pyx

```diff
@@ -579,59 +582,66 @@ cpdef array_to_datetime(ndarray[object] values, str errors='raise',
                         iresult[i] = NPY_NAT
                         continue
 
+                    string_to_dts_failed = _string_to_dts(
+                        val, &dts, &out_local,
+                        &out_tzoffset
+                    )
                     try:
-                        _string_to_dts(val, &dts, &out_local, &out_tzoffset)
-                    except ValueError:
-                        # A ValueError at this point is a _parsing_ error
-                        # specifically _not_ OutOfBoundsDatetime
-                        if _parse_today_now(val, &iresult[i]):
-                            continue
-                        elif require_iso8601:
-                            # if requiring iso8601 strings, skip trying
-                            # other formats
-                            if is_coerce:
-                                iresult[i] = NPY_NAT
+                        if string_to_dts_failed:
```

Comment:
this try/except seems uncessary now, yes?

Reply:
You're rigth.

Reply:
It is actually not exactly so. We're still relying on exceptions when trying to get a string buffer.
This means we've moved to checking return code for exceptions which were used in _normal_ code flow, but there are still possibilities of exceptions in, well, exceptional code flow (like `OutOfMemoryException` when generating a UTF8 representation for a widechar Unicode string).